JavaScript/packages/recognizers-date-time/src/dateTime/baseMerged.ts

```diff
@@ -168,26 +168,33 @@ export class BaseMergedExtractor implements IDateTimeExtractor {
         let lastEnd = 0;
         ers.forEach(er => {
             let beforeStr = source.substr(lastEnd, er.start).toLowerCase();
+            let isSuccess = false;
             let before = this.hasTokenIndex(beforeStr.trim(), this.config.beforeRegex);
             if (before.matched) {
                 let modLength = beforeStr.length - before.index;
                 er.length += modLength;
                 er.start -= modLength;
                 er.text = source.substr(er.start, er.length);
+                isSuccess = true;
             }
-            let after = this.hasTokenIndex(beforeStr.trim(), this.config.afterRegex);
-            if (after.matched) {
-                let modLength = beforeStr.length - after.index;
-                er.length += modLength;
-                er.start -= modLength;
-                er.text = source.substr(er.start, er.length);
+            if(!isSuccess){
```

Comment:
Nit: Please add whitespaces to match the code style. Also, I suggest adding a blank line between two consecutive ifs.